snuba/datasets/entities/discover.py

```diff
@@ -251,6 +258,67 @@ def attempt_map(
             return None
 
 
+@dataclass(frozen=True)
+class DefaultNoneFunctionMapper(FunctionCallMapper):
+    """
+    If a function is being called on a column that doesn't exist, or is being
+    called on NULL, change the entire function to be NULL.
+    """
+
+    columns: ColumnSet
+
+    def attempt_map(
+        self,
+        expression: FunctionCall,
+        children_translator: SnubaClickhouseStrictTranslator,
+    ) -> Optional[FunctionCall]:
+        valid_parameters: List[Expression] = []
+        column_match = ColumnMatch(Or([StringMatch(c.flattened) for c in self.columns]))
+        # This is called recursively, so use this match to collapse ifNull calls in the case of
+        # unexpected columns being wrapped in more than one function.
+        function_match = FunctionCallMatch(
+            StringMatch("ifNull"), (LiteralMatch(), LiteralMatch())
+        )
+        for param in expression.parameters:
+            cmatch = column_match.match(param)
+            if cmatch is not None:
+                assert isinstance(param, Column)
+                valid_parameters.append(
+                    Literal(
+                        alias=param.alias
+                        or qualified_column(param.column_name, param.table_name or ""),
+                        value=None,
+                    )
+                )
+                continue
+
+            fmatch = function_match.match(param)
+            if fmatch is not None:
+                valid_parameters.append(Literal(param.alias, None))
+                continue
+
+            valid_parameters.append(param)
```

Comment:
If you are looking for parameters that would be translated to NULL, you shoulld use the `children_translator` argument you receive in this method, and just go ahead to translate them. That is going to be more robust since it will not create an implicit coupling between this code and the translation rules applied elsewhere to the columns.
Relying on the translator to translate the children would work no matter how the the columns not present on the table you are querying are translated.